lib/srv/sess.go

```diff
@@ -761,8 +765,16 @@ func (s *session) Close() error {
 				}
 			}
 
+			// Complete the session recording
 			if s.recorder != nil {
-				s.recorder.Close(s.serverCtx)
+				if err := s.recorder.Complete(s.serverCtx); err != nil {
+					s.log.WithError(err).Warn("Failed to close recorder.")
+				}
+				select {
+				case <-s.recorder.Done():
+				case <-time.After(time.Second * 30):
```

Comment:
30s seems like a long time to wait - under what circumstances is this called?

Reply:
This is called during session cleanup, after the client's session is already closed etc. This select statement is intended to wait for the session uploader to close/complete before sending the `SessionStateTerminated` and potentially triggering the upload completer process - triggering the race condition. I'll add a comment.

Reply:
Just wondering what negative effects this might have.. Close used to finish immediately, now it can take up to 30s.

Reply:
Fair enough. I don't think this is needed after all since terminating the session won't instantly expire session trackers now.